src/ensembl/src/content/app/custom-download/components/checkbox-with-selects/CheckboxWithSelects.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+import Select, { Option } from 'src/shared/select/Select';
+import styles from './CheckboxWithSelects.scss';
+
+import ImageButton from 'src/shared/image-button/ImageButton';
+import { ReactComponent as AddIcon } from 'static/img/browser/zoom-in.svg';
+import { ReactComponent as RemoveIcon } from 'static/img/shared/clear.svg';
+
+type Props = {
+  selectOptions: Option[];
+  label: string;
+  selectedOptions: string[];
+  onChange: (selectedOptions: []) => void;
+};
+
+const CheckboxWithSelects = (props: Props) => {
+  const [checkedStatus, setCheckedStatus] = useState(false);
+  const [shouldShowExtraOption, setShowExtraOption] = useState(false);
+
+  useEffect(() => {
+    if (props.selectedOptions.length > 0) {
+      setCheckedStatus(true);
+    }
+  }, []);
+
+  const handleCheckboxOnChange = useCallback(
+    (checkedStatus: boolean) => {
+      setCheckedStatus(checkedStatus);
+      props.onChange([]);
+    },
+    [props.selectedOptions]
+  );
```

Comment:
does it have to be wrapped in useCallback updated by a change in props.selectOptions?

Reply:
Removed the useCallback wrapper.